app/graphql/mutations/update_mutation.rb

```diff
@@ -0,0 +1,31 @@
+module Mutations
+  class UpdateMutation < BaseMutation
+    field :errors, [Types::UserError], null: false
+
+    def resolve(params)
+      object = ForemanGraphqlSchema.object_from_id(params[:id], context)
+
+      object.assign_attributes(params.except(:id))
+
+      authorize!(object, :edit)
+
+      User.as(context[:current_user]) do
+        errors = if object.save
+                   []
+                 else
+                   object.errors.map do |attribute, message|
```

Comment:
Maybe extract to stay DRY as both `UpdateMutation` and `CreateMutation` use same error handling.

Reply:
Extracted to `save_object` method.